### DISCUSSION TITLE
It works like a charm - Can you help me understand the modbus communication?

### DISCUSSION THREAD

**Discussion by OJBender** [pail23/stiebel_eltron_isg_component#169] [Q&A]
@pail23 THANKS SO MUCH FOR THAT! I am the happiest man in our town with this integration. Finally with PV Überschuss I can start the SG Ready program.   I know this is slightly OFF TOPIC as it does not directly address your great integration. But I am sure you could answer my question with a wink of an eye.

<br>
Would you give me ONE hint to understand the modbus communication? I wanted to use NodeRED with which I was successful reading SMA inverters and energy meters. The Stiebel ISG seems different.
Stiebel Eltron says for example "READ INPUT REGISTER" 5001 returns either 1,2,3 or 4. 
When I send the FC2 Read Input Status to 5001, I get no response at all. I was thinking to send it maybe in hex (0x1389), then added and subtracted 1 from the address with no success.

Would you tell me where I am wrong? What are you sending for example to the ISG to read that input register 5001 or a holding register 1004? Don't hate me for asking this. 

Yours
OJ

**Comment by pail23** [comment]
What component does is reading the input registers on the modbus. Example:
[`        inverter_data = self.read_input_registers(slave=1, address=500, count=96)`](https://github.com/pail23/stiebel_eltron_isg_component/blob/6c5f577c06e9241452ee81837bc2d7607a5640d2/custom_components/stiebel_eltron_isg/wpm_coordinator.py#L149C1-L149C82)

This line returns the value of the input registers 501 - 507 as described in the [Stiebel Eltron modbus documentation](https://www.stiebel-eltron.ch/content/dam/ste/ch/de/downloads/kundenservice/smart-home/Modbus/Modbus%20Bedienungsanleitung.pdf). Please be aware that the numbering in documentation is off by 1 (you need to send 500 in the code when you want to read register 501 as described in the documentation)

Register 5001 does not exist. Do you mean 501?
